server/services/user.js

```diff
@@ -90,6 +90,19 @@ function removeGroup(user, groupName) {
     .then(validatedName => user.removeGroup(validatedName));
 }
 
+function getGroupsForUser(user) {
+  return new Promise((resolve, reject) => {
+    User.findOne(user)
+    .populate('activeGroups')
+    .exec((err, u) => {
```

Comment:
>Queries do return a thenable, but if you need a real Promise you should use the exec method

Also if you take a look in the [docs](http://mongoosejs.com/docs/promises.html) it says this
>If you need a fully-fledged promise, use the .exec() function.

Just to make sure our promise is set to the correct Promise type and not mpromise as is default with mongoose, I ran the following test which passed.
```
const q = User.findOne({});
expect(q.exec().constructor).to.equal(global.Promise);
```